packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -215,32 +214,21 @@ export const prepareLiquidityPoolKit = (zone, zcf, USDC, tools) => {
             shareWorth,
             encumberedBalance,
           );
-
-          const fromSeatAllocation = fromSeat.getCurrentAllocation();
-          // Validate allocation equals amounts and Principal <= encumberedBalance
+          // Validate Principal <= encumberedBalance and produce poolStats after
           const post = repayCalc(
             shareWorth,
-            fromSeatAllocation,
-            amounts,
+            split,
             encumberedBalance,
             poolStats,
           );
 
-          const { ContractFee, ...rest } = amounts;
-
           // COMMIT POINT
           // UNTIL #10684: ability to terminate an incarnation w/o terminating the contract
-          zcf.atomicRearrange(
-            harden([
-              [
-                fromSeat,
-                poolSeat,
-                rest,
-                { USDC: add(amounts.PoolFee, amounts.Principal) },
-              ],
-              [fromSeat, feeSeat, { ContractFee }, { USDC: ContractFee }],
-            ]),
-          );
+          zcf.atomicRearrange([
```

Comment:
I could swear zcf throws when this `harden(...)` is missing. But maybe only the individual parts need to be hardened?

Reply:
done.

Do you believe that even though tests all passed?

Reply:
I think `fast-usdc.contract.test.ts` is sufficiently high fidelity that if the tests pass, the code would work on chain without `harden(...)`.